src/prefect/cli/deploy.py

```diff
@@ -500,118 +552,66 @@ async def _run_single_deploy(
     base_deploy["schedule"] = schedule
 
     # prepare the pull step
+    pull_steps = base_deploy.get(
+        "pull", project.get("pull")
+    ) or await _generate_default_pull_action(
+        app.console,
+        base_deploy=base_deploy,
+    )
     pull_steps = apply_values(pull_steps, step_outputs)
 
-    async with prefect.get_client() as client:
-        flow_id = await client.create_flow_from_name(base_deploy["flow_name"])
-
-        if base_deploy["work_pool"]["name"]:
-            try:
-                work_pool = await client.read_work_pool(
-                    base_deploy["work_pool"]["name"]
-                )
-
-                # dont allow submitting to prefect-agent typed work pools
-                if work_pool.type == "prefect-agent":
-                    if not is_interactive() or ci:
-                        raise ValueError(
-                            "Cannot create a project-style deployment with work pool of"
-                            " type 'prefect-agent'. If you wish to use an agent with"
-                            " your deployment, please use the `prefect deployment"
-                            " build` command."
-                        )
-                    app.console.print(
-                        "You've chosen a work pool with type 'prefect-agent' which"
-                        " cannot be used for project-style deployments. Let's pick"
-                        " another work pool to deploy to."
-                    )
-                    base_deploy["work_pool"]["name"] = await _prompt_select_work_pool(
-                        app.console,
-                        client=client,
-                    )
-            except ObjectNotFound:
-                app.console.print(
-                    (
-                        "\nThis deployment references a work pool that does not exist."
-                        " This means no worker will be able to pick up its runs. You"
-                        " can create a work pool in the Prefect UI."
-                    ),
-                    style="red",
-                )
-        else:
-            if not is_interactive() or ci:
-                raise ValueError(
-                    "A work pool is required to deploy this flow. Please specify a work"
-                    " pool name via the '--pool' flag or in your deployment.yaml file."
-                )
-            base_deploy["work_pool"]["name"] = await _prompt_select_work_pool(
-                console=app.console, client=client
-            )
```

Comment:
Moved this earlier in the flow so that user's select their work pool __before__ the deployment actions are run.